### PR TITLE
Add a "serve" command to "closure-util" binary

### PR DESCRIPTION
**Pull request by elemoine**
This PR adds a "serve" command to the "closure-util" binary. This command starts a development server that can be used to serve code using `goog.provide` and `goog.require`.

This is how the `serve` command is used:

```
closure-util serve config.json
```

And this is what the JSON config file (`config.json`) looks like:

```json
{
  "lib": [
    "path/to/my/js/files/**.js"
  ],
  "serve": {
    "port": 3001
  }
}
```

That configuration file  can be the same file as the one used with the `build` command (with an additional `"serve"` section).

Requires https://github.com/openlayers/closure-util/pull/52. (now closed)

### CODE REVIEW

**Comment by elemoine** [comment]
Now rebased on the current master.

**Comment by elemoine** [comment]
I just updated the description of the issue. Please review.

File: lib/build.js [hunk 1]
```diff
@@ -22,7 +22,35 @@ log.level = config.get('log_level');
...
+}
+
+
+/**
+ * Assert that a provided config object is valid.
+ * @param {Object} config Build configuration object.
+ * @param {function(Error, Object)} callback Called with the configuration
+ *     object or any error.
+ */
+function _assertValidConfig(config, callback) {
```
Comment (tschaub) [inline lib/build.js]: I'm sure you know, but the `_privateStyle` names are not really necessary.  Only properties explicitly assigned to `exports` or `module.exports` are available outside this module.

**Comment by tschaub** [comment]
Looks great @elemoine.  One minor comment about the `_namingConvention`, but that is a stylistic thing (and not yet enforced).

**Comment by elemoine** [comment]
> Looks great @elemoine. One minor comment about the _namingConvention, but that is a stylistic thing (and not yet enforced).

I'll remove the underscore before merging. Thanks for the review.

**Pull request merged**